hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/WriteFileThroughputBenchmark.java

```diff
@@ -0,0 +1,232 @@
+
+  private long expectedIoTimeNs;
+
+  private Timer timer;
+
+  private final ThreadLocal<FileSystem> threadLocalFileSystem =
+      ThreadLocal.withInitial(this::createFS);
+
+  public static final Logger LOG =
+      LoggerFactory.getLogger(WriteFileThroughputBenchmark.class);
+
+
+  public Void call() throws Exception {
+
+    init();
+
+    // We cannot have both Hsync and Hflush set to TRUE at the same time
+    if (hSync && hFlush) {
+      if (hSync == hFlush) {
+        LOG.info("Both Hsync and Hflush cannot be set to TRUE");
+        System.exit(1);
+      }
+    }
+
+    LOG.info("NumFiles=" + getTestNo());
+    LOG.info("Total FileSize=" + fileSize);
+    LOG.info("BlockSize=" + blockSize);
+    LOG.info("BufferSize=" + bufferSize);
+    LOG.info("Replication=" + replication);
+    LOG.info("Threads=" + getThreadNo());
+    LOG.info("URI Scheme Used=" + uri.getScheme());
+    if (hSync) {
+      LOG.info("Hsync after every write= True");
+    } else if (hFlush) {
+      LOG.info("Hflush after every write= True");
+    }
+
+    // Initialize the configuration variable with
+    // OzoneFS configuration
+    configuration = createOzoneConfiguration();
+
+    //Constructs a URI object by parsing the given string rootPath
+    uri = URI.create(rootPath);
+
+    // Disabling the file system cache
+    String disableCacheName = String.format("fs.%s.impl.disable.cache",
+        uri.getScheme());
+    LOG.info("Disabling FS cache: " + disableCacheName);
+    configuration.setBoolean(disableCacheName, true);
+
+    Path file = new Path(rootPath + "/" +
+        generateObjectName(0));
+    FileSystem fileSystem =  FileSystem.get(configuration);
+    fileSystem.mkdirs(file.getParent());
+    // Checks whether output directory exists
+    ensureOutputDirExists(createFS(), file);
+
+    // Initialize the size of the file to be written in Bytes
+    long filesizeinBytes = fileSize * 1_000_000_000;
+    contentGenerator = new ContentGenerator(filesizeinBytes,
+        bufferSize, hSync, hFlush);
+
+    // Initializing the time it should take to write a file
+    expectedIoTimeNs =
+        (isThrottled ? (((long) bufferSize * 1_000_000_000) / throttle)
+            : 0);
+
+    timer = getMetrics().timer("file-create");
+
+    runTests(this::createFile);
+
+    return null;
+  }
+
+
+  private void createFile(long counter) throws Exception {
+    Path file = new Path(rootPath + "/" + generateObjectName(counter));
+    FileSystem fileSystem = threadLocalFileSystem.get();
+
+    final long ioStartTimeNs = (isThrottled ? System.nanoTime() : 0);
+
+    timer.time(() -> {
+      try (FSDataOutputStream outputStream = fileSystem.create(file,
+          false, bufferSize, replication, blockSize)) {
+        contentGenerator.write(outputStream);
+
+        // Enforcing throttle delay
+        final long ioEndTimeNs = (isThrottled ? System.nanoTime() : 0);
+        enforceThrottle(ioEndTimeNs - ioStartTimeNs, expectedIoTimeNs);
```

Comment:
Can you please explain the need for throttling in a performance test?

If it is really necessary, shouldn't it happen outside of the `timer.time()` call?

Reply:
The original FsPerfTest tool was created to measure both throughput and latency. The throttling option would be more relevant in the context of latency. We can drop the throttling for now as this PR does not support measuring latency. But later, throttling can be added back when latency metrics are added.

Link to the FsPerfTest tool :-https://github.com/arp7/FsPerfTest/blob/master/src/main/java/net/arp7/FsPerfTest/WriteFile.java